src/core/fileinfojob.cpp

```diff
@@ -23,9 +31,15 @@ void FileInfoJob::exec() {
 
             // Reuse the same dirPath object when the path remains the same (optimize for files in the same dir)
             auto dirPath = commonDirPath_.isValid() ? commonDirPath_ : path.parent();
-            std::shared_ptr<const FileInfo> fileInfo = std::make_shared<FileInfo>(inf, dirPath);
-            results_.push_back(fileInfo);
-            Q_EMIT gotInfo(path, fileInfo);
+            auto fileInfo = std::make_shared<FileInfo>(inf, dirPath);
+
+            if(hasCutFiles && cutFilesHashSet_->count(fileInfo->path().hash()))
+                fileInfo->bindCutFiles(cutFilesHashSet_);
+
+            auto fileInfoConst = std::const_pointer_cast<const FileInfo>(fileInfo);
```

Comment:
I see now, maybe I should have created `shared_ptr` only here, then `const_pointer_cast` can be ommited - _Can fix.._